java/ql/src/experimental/Security/CWE/CWE-094/GroovyInjectionBad.java

```diff
@@ -0,0 +1,49 @@
+public class GroovyInjection extends HttpServlet {
+
+    protected void doGet(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+        try {
+            String script = request.getParameter("script");
+            final GroovyClassLoader classLoader = new GroovyClassLoader();
+            Class groovy = classLoader.parseClass(script);
+            GroovyObject groovyObj = (GroovyObject) groovy.newInstance();
+
+        } catch (Exception e) {
+            // Ignore
+        }
+    }
+
+    protected void doPost(HttpServletRequest request, HttpServletResponse response)
```

Comment:
Name the test methods `injectionViaClassLoader`, `injectionViaEval` etc (they can still take an `HttpServletRequest` as a parameter to act as an untrusted source) -- the HTTP method is irrelevant here